src/config/config.py

```diff
@@ -0,0 +1,123 @@
+"""Module to handle configuration application configuration parameters."""
+import configparser
+import logging
+import os
+from pathlib import Path
+from typing import Final
+
+logger = logging.getLogger(__name__)
+
+CONFIG_FILE_NAME: Final[str] = "config.ini"
+CONFIG_DIRECTORY_NAME: Final[str] = "nina"
+
+
+def find_config_directory() -> Path:  # Pragma: no cover
+    """Get configuration directory.
+
+    Checks the operating system if it is windows or unix based. Uses enviroment
+    variables 'LOCALAPPDATA', 'XDG_CONFIG_HOME', and 'HOME' to determine
+    applications config directory. Defaults to the current working directory
+    if none of the above enviroment variables are set.
+
+    Returns
+    -------
+    Path    :
+        Path object to the configuration directory.
+    """
+    if os.name == "nt" and "LOCALAPPDATA" in os.environ:
```

Comment:
Can maybe look into mocking `os.path` so that you can test the function.

https://gist.github.com/tomschr/d644d924793e0588fb04f1ea74dabbc1
https://pypi.org/project/pytest-mock/

Reply:
Not sure this is that simple to mock since we are working with `Path` objects. When trying to mock the `os.name` it also messes with the internal logic of `pathlib`. See the following snippet:
```py
from unittest.mock import patch

@patch.dict(os.environ, {"LOCALAPPDATA": "/home/kad"}, clear=True)
@patch("os.name", "nt")
def test_config_dir_windows():
    logger.info(config.find_config_directory())
```
Results in:
```
================================================================ FAILURES =================================================================
_________________________________________________________ test_config_dir_windows _________________________________________________________

    @patch.dict(os.environ, {"LOCALAPPDATA": "test"}, clear=True)
    @patch("os.name", "nt")
    def test_config_dir_windows():
>       logger.info(config.find_config_directory())

tests/unit/config/test_config.py:43: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src/config/config.py:30: in find_config_directory
    confighome = Path(os.environ["LOCALAPPDATA"])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'pathlib.WindowsPath'>, args = ('test',), kwargs = {}, self = WindowsPath('test')

    def __new__(cls, *args, **kwargs):
        if cls is Path:
            cls = WindowsPath if os.name == 'nt' else PosixPath
        self = cls._from_parts(args, init=False)
        if not self._flavour.is_supported:
>           raise NotImplementedError("cannot instantiate %r on your system"
                                      % (cls.__name__,))
E           NotImplementedError: cannot instantiate 'WindowsPath' on your system

/usr/lib64/python3.8/pathlib.py:1044: NotImplementedError
========================================================= short test summary info =========================================================
FAILED tests/unit/config/test_config.py::test_config_dir_windows - NotImplementedError: cannot instantiate 'WindowsPath' on your system
======================================================= 1 failed, 5 passed in 0.07s =======================================================
```

Reply:
According to the [pathlib documentation](https://docs.python.org/3/library/pathlib.html). You cannot instantiate a PosixPath on a Windows machine, and the same for a WindowsPath on a unix machine. There exists "Pure" versions of either in case you need to work with the other, but they do not provide any I/O functions. Which is what is needed to test this function. Since we are running integration tests on both platforms here on GitHub, i don't think this that big of an issue.

Reply:
I plan on use [Click](https://click.palletsprojects.com/en/8.0.x/index.html) for ui in the future.  It’s bundled with Flask.

For reference, you can look at how they’ve [added](https://github.com/pallets/click/blob/9da166957f5848b641231d485467f6140bca2bc0/src/click/utils.py#L412) same logic.  Initialliy I tried finding the tests, but could not find them.